### PR TITLE
Subordinate container runtime

### PR DESCRIPTION
**Pull request by joedborg**
Succeeds https://github.com/charmed-kubernetes/layer-calico/pull/35

### CODE REVIEW

**Comment by joedborg** [comment]
Hold for merge conflict.

File: metadata.yaml [hunk 1]
```diff
@@ -38,3 +39,7 @@ resources:
     type: file
     filename: calico-upgrade.tar.gz
     description: 'calico-upgrade tool for arm64'
+  calico-node-image:
```
Comment (Cynerva) [inline metadata.yaml]: Remind me, this resource was added as a workaround for the issue where `ctr image pull` does not properly use the *_proxy configs from the containerd subordinate, correct?

The resource will need to be added to our build automation:
* https://github.com/charmed-kubernetes/layer-calico/blob/master/build-calico-resource.sh
* https://github.com/charmed-kubernetes/jenkins/blob/master/jobs/build-charms/resource-spec.yaml

Comment (joedborg) [reply metadata.yaml]: That's correct @Cynerva, so we'll have to add it as you say.

**Comment by Cynerva** [comment]
Missing resource build automation aside, this LGTM.

**Pull request merged**